client/src/pages/Home.jsx

```diff
@@ -1,7 +1,19 @@
+import { useState } from 'react'
+import { UserAPI } from '../apis/UserAPI'
+import SearchBar from '../components/SearchBar'
+
 const Home = () => {
+  const [users, setUsers] = useState([])
+
+  const handleSearch = async query => {
+    const results = query ? await UserAPI.search(query) : await UserAPI.getAll()
```

Comment:
Are we thinking of going with @tanstack/react-query version 5 for our project? I'm on the fence about whether caching would really help us out here since it looks like every query we're dealing with is pretty unique. But maybe, there's some value in caching the searches we've done before? It could save us from having to fetch new data every single time. What do you think?

Reply:
Will it make life easier for pagination of results for the same query?

Reply:
Can add caching and pagination, but believe these can be lower priority and done later in the project. I'll create a ticket for these and toss it into the backlog.